src/controllers/handlerFactory.ts

```diff
@@ -161,12 +155,23 @@ export const getAllEntities = <ModelInterface>(
             ],
           }
         : {};
+      const query = req.query as RequestQueryString;
 
-      const documents: ModelInterface[] = await ModelEntity.find(filters);
+      const features = new APIFeatures(ModelEntity, query)
+        .filter()
+        .sort()
+        .select()
+        .paginate();
+
+      const documents: ModelInterface[] = await features.query;
+      const totalDocuments = await ModelEntity.countDocuments(
+        features.query.getFilter()
+      );
 
       res.status(200).json({
         status: 'success',
         results: documents.length,
+        totalDocuments,
```

Comment:
That totalDocument comes in json format? total likes: 5? or count 

if it is not the case, you can put it total or count : totalDocuments.